### PR TITLE
Plasma 5.27

### PR DESCRIPTION
**Pull request by K900**
###### Description of changes

https://kde.org/announcements/plasma/5/5.27.0/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by K900** [comment]
lxqt-config fails due to https://github.com/lxqt/lxqt-config/issues/903

File: pkgs/development/libraries/kuserfeedback/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  lib,
+  mkDerivation,
+  fetchurl,
+  extra-cmake-modules,
+  qttools,
+}:
+mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/kuserfeedback/default.nix:8]:
```suggestion
{ lib
, mkDerivation
, fetchurl
, extra-cmake-modules
, qttools
}:

mkDerivation rec {
```

File: pkgs/development/libraries/kuserfeedback/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  };
+
+  nativeBuildInputs = [ extra-cmake-modules qttools ];
+
+  meta = with lib; {
+    license = [ licenses.mit ];
+    maintainers = [ maintainers.k900 ];
+    description = ''
+      Framework for collecting user feedback for apps via telemetry and surveys
+    '';
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/kuserfeedback/default.nix:24]:
```suggestion
    description = "Framework for collecting user feedback for apps via telemetry and surveys";
```

**Comment by K900** [comment]
Result of `nixpkgs-review pr 211767` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>12 packages failed to build:</summary>
  <ul>
    <li>gmic-qt-krita</li>
    <li>hotspot</li>
    <li>kde-rounded-corners</li>
    <li>libsForQt5.applet-window-buttons</li>
    <li>libsForQt5.buho</li>
    <li>libsForQt5.clip</li>
    <li>libsForQt5.communicator</li>
    <li>libsForQt5.nota</li>
    <li>libsForQt5.pix</li>
    <li>libsForQt5.shelf</li>
    <li>libsForQt5.vvave</li>
    <li>lxqt.lxqt-config</li>
  </ul>
</details>
<details>
  <summary>356 packages built:</summary>
  <ul>
    <li>ajour</li>
    <li>amarok</li>
    <li>bcompare</li>
    <li>calamares</li>
    <li>calamares-nixos</li>
    <li>crow-translate</li>
    <li>digikam</li>
    <li>fcitx5-configtool</li>
    <li>fcitx5-with-addons</li>
    <li>goverlay</li>
    <li>haruna</li>
    <li>heaptrack</li>
    <li>heroic</li>
    <li>k4dirstat</li>
    <li>kaffeine</li>
    <li>kaidan</li>
    <li>kbibtex</li>
    <li>kcollectd</li>
    <li>kdbg</li>
    <li>kdiff3</li>
    <li>kgraphviewer</li>
    <li>kile</li>
    <li>kio-fuse</li>
    <li>kmplayer</li>
    <li>kmymoney</li>
    <li>kotatogram-desktop</li>
    <li>kotatogram-desktop-with-webkit</li>
    <li>kphotoalbum</li>
    <li>krename</li>
    <li>krita</li>
    <li>kronometer</li>
    <li>krunner-pass</li>
    <li>krusader</li>
    <li>ksmoothdock</li>
    <li>kstars</li>
    <li>ktimetracker</li>
    <li>kup</li>
    <li>kwalletcli</li>
    <li>latte-dock</li>
    <li>libreoffice-qt</li>
    <li>libsForQt5.akonadi</li>
    <li>libsForQt5.akonadi-calendar</li>
    <li>libsForQt5.akonadi-calendar-tools</li>
    <li>libsForQt5.akonadi-contacts</li>
    <li>libsForQt5.akonadi-import-wizard</li>
    <li>libsForQt5.akonadi-mime</li>
    <li>libsForQt5.akonadi-notes</li>
    <li>libsForQt5.akonadi-search</li>
    <li>libsForQt5.akonadiconsole</li>
    <li>libsForQt5.akregator</li>
    <li>libsForQt5.alkimia</li>
    <li>libsForQt5.angelfish</li>
    <li>libsForQt5.ark</li>
    <li>libsForQt5.aura-browser</li>
    <li>libsForQt5.baloo</li>
    <li>libsForQt5.baloo-widgets</li>
    <li>libsForQt5.bismuth</li>
    <li>libsForQt5.bluedevil</li>
    <li>libsForQt5.bomber</li>
    <li>libsForQt5.booth</li>
    <li>libsForQt5.bovo</li>
    <li>libsForQt5.breeze-grub</li>
    <li>libsForQt5.breeze-gtk</li>
    <li>libsForQt5.breeze-plymouth</li>
    <li>libsForQt5.breeze-qt5</li>
    <li>libsForQt5.calendarsupport</li>
    <li>libsForQt5.colord-kde</li>
    <li>libsForQt5.discover</li>
    <li>libsForQt5.dolphin</li>
    <li>libsForQt5.dolphin-plugins</li>
    <li>libsForQt5.dragon</li>
    <li>libsForQt5.elisa</li>
    <li>libsForQt5.eventviews</li>
    <li>libsForQt5.falkon</li>
    <li>libsForQt5.ffmpegthumbs</li>
    <li>libsForQt5.filelight</li>
    <li>libsForQt5.flatpak-kcm</li>
    <li>libsForQt5.frameworkintegration</li>
    <li>libsForQt5.granatier</li>
    <li>libsForQt5.grantleetheme</li>
    <li>libsForQt5.gwenview</li>
    <li>libsForQt5.incidenceeditor</li>
    <li>libsForQt5.index</li>
    <li>libsForQt5.juk</li>
    <li>libsForQt5.k3b</li>
    <li>libsForQt5.kaccounts-integration</li>
    <li>libsForQt5.kaccounts-providers</li>
    <li>libsForQt5.kactivities</li>
    <li>libsForQt5.kactivities-stats</li>
    <li>libsForQt5.kactivitymanagerd</li>
    <li>libsForQt5.kaddressbook</li>
    <li>libsForQt5.kalarm</li>
    <li>libsForQt5.kalendar</li>
    <li>libsForQt5.kalzium</li>
    <li>libsForQt5.kamoso</li>
    <li>libsForQt5.kapman</li>
    <li>libsForQt5.kapptemplate</li>
    <li>libsForQt5.kate</li>
    <li>libsForQt5.katomic</li>
    <li>libsForQt5.kblackbox</li>
    <li>libsForQt5.kblocks</li>
    <li>libsForQt5.kbookmarks</li>
    <li>libsForQt5.kbounce</li>
    <li>libsForQt5.kbreakout</li>
    <li>libsForQt5.kcachegrind</li>
    <li>libsForQt5.kcalc</li>
    <li>libsForQt5.kcalutils</li>
    <li>libsForQt5.kcharselect</li>
    <li>libsForQt5.kclock</li>
    <li>libsForQt5.kcmutils</li>
    <li>libsForQt5.kcolorchooser</li>
    <li>libsForQt5.kconfigwidgets</li>
    <li>libsForQt5.kdav</li>
    <li>libsForQt5.kde-cli-tools</li>
    <li>libsForQt5.kde-gtk-config</li>
    <li>libsForQt5.kde2-decoration</li>
    <li>libsForQt5.kdeclarative</li>
    <li>libsForQt5.kdeconnect-kde</li>
    <li>libsForQt5.kdecoration</li>
    <li>libsForQt5.kdegraphics-mobipocket</li>
    <li>libsForQt5.kdegraphics-thumbnailers</li>
    <li>libsForQt5.kdelibs4support</li>
    <li>libsForQt5.kdenetwork-filesharing</li>
    <li>libsForQt5.kdenlive</li>
    <li>libsForQt5.kdepim-addons</li>
    <li>libsForQt5.kdepim-runtime</li>
    <li>libsForQt5.kdeplasma-addons</li>
    <li>libsForQt5.kdesignerplugin</li>
    <li>libsForQt5.kdev-php</li>
    <li>libsForQt5.kdev-python</li>
    <li>libsForQt5.kdevelop</li>
    <li>libsForQt5.kdevelop-pg-qt</li>
    <li>libsForQt5.kdevelop-unwrapped</li>
    <li>libsForQt5.kdf</li>
    <li>libsForQt5.kdialog</li>
    <li>libsForQt5.kdiamond</li>
    <li>libsForQt5.keditbookmarks</li>
    <li>libsForQt5.kfind</li>
    <li>libsForQt5.kfloppy</li>
    <li>libsForQt5.kgamma5</li>
    <li>libsForQt5.kgeography</li>
    <li>libsForQt5.kget</li>
    <li>libsForQt5.kgpg</li>
    <li>libsForQt5.kguiaddons</li>
    <li>libsForQt5.khelpcenter</li>
    <li>libsForQt5.khotkeys</li>
    <li>libsForQt5.khtml</li>
    <li>libsForQt5.kiconthemes</li>
    <li>libsForQt5.kidentitymanagement</li>
    <li>libsForQt5.kidletime</li>
    <li>libsForQt5.kig</li>
    <li>libsForQt5.kigo</li>
    <li>libsForQt5.killbots</li>
    <li>libsForQt5.kimap</li>
    <li>libsForQt5.kinfocenter</li>
    <li>libsForQt5.kinit</li>
    <li>libsForQt5.kio</li>
    <li>libsForQt5.kio-extras</li>
    <li>libsForQt5.kio-gdrive</li>
    <li>libsForQt5.kipi-plugins</li>
    <li>libsForQt5.kldap</li>
    <li>libsForQt5.kleopatra</li>
    <li>libsForQt5.klettres</li>
    <li>libsForQt5.klines</li>
    <li>libsForQt5.kmag</li>
    <li>libsForQt5.kmahjongg</li>
    <li>libsForQt5.kmail</li>
    <li>libsForQt5.kmail-account-wizard</li>
    <li>libsForQt5.kmailtransport</li>
    <li>libsForQt5.kmediaplayer</li>
    <li>libsForQt5.kmenuedit</li>
    <li>libsForQt5.kmines</li>
    <li>libsForQt5.kmix</li>
    <li>libsForQt5.kmousetool</li>
    <li>libsForQt5.kmplot</li>
    <li>libsForQt5.knavalbattle</li>
    <li>libsForQt5.knetwalk</li>
    <li>libsForQt5.knewstuff</li>
    <li>libsForQt5.knights</li>
    <li>libsForQt5.knotes</li>
    <li>libsForQt5.knotifyconfig</li>
    <li>libsForQt5.koko</li>
    <li>libsForQt5.kolf</li>
    <li>libsForQt5.kollision</li>
    <li>libsForQt5.kolourpaint</li>
    <li>libsForQt5.kompare</li>
    <li>libsForQt5.konqueror</li>
    <li>libsForQt5.konquest</li>
    <li>libsForQt5.konsole</li>
    <li>libsForQt5.kontact</li>
    <li>libsForQt5.kontactinterface</li>
    <li>libsForQt5.konversation</li>
    <li>libsForQt5.korganizer</li>
    <li>libsForQt5.kparts</li>
    <li>libsForQt5.kpat</li>
    <li>libsForQt5.kpimtextedit</li>
    <li>libsForQt5.kpipewire</li>
    <li>libsForQt5.kpmcore</li>
    <li>libsForQt5.kproperty</li>
    <li>libsForQt5.kpurpose</li>
    <li>libsForQt5.krdc</li>
    <li>libsForQt5.kreversi</li>
    <li>libsForQt5.krfb</li>
    <li>libsForQt5.krohnkite</li>
    <li>libsForQt5.kross</li>
    <li>libsForQt5.kruler</li>
    <li>libsForQt5.krunner</li>
    <li>libsForQt5.krunner-symbols</li>
    <li>libsForQt5.kscreen</li>
    <li>libsForQt5.kscreenlocker</li>
    <li>libsForQt5.kshisen</li>
    <li>libsForQt5.ksmtp</li>
    <li>libsForQt5.kspaceduel</li>
    <li>libsForQt5.ksquares</li>
    <li>libsForQt5.ksshaskpass</li>
    <li>libsForQt5.ksudoku</li>
    <li>libsForQt5.ksystemlog</li>
    <li>libsForQt5.ksystemstats</li>
    <li>libsForQt5.kteatime</li>
    <li>libsForQt5.ktexteditor</li>
    <li>libsForQt5.ktextwidgets</li>
    <li>libsForQt5.ktimer</li>
    <li>libsForQt5.ktnef</li>
    <li>libsForQt5.ktorrent</li>
    <li>libsForQt5.ktouch</li>
    <li>libsForQt5.ktrip</li>
    <li>libsForQt5.kturtle</li>
    <li>libsForQt5.kuserfeedback</li>
    <li>libsForQt5.kwallet</li>
    <li>libsForQt5.kwallet-pam</li>
    <li>libsForQt5.kwalletmanager</li>
    <li>libsForQt5.kwave</li>
    <li>libsForQt5.kwayland</li>
    <li>libsForQt5.kwayland-integration</li>
    <li>libsForQt5.kweather</li>
    <li>libsForQt5.kwin</li>
    <li>libsForQt5.kwin-dynamic-workspaces</li>
    <li>libsForQt5.kwin-tiling</li>
    <li>libsForQt5.kwrited</li>
    <li>libsForQt5.kxmlgui</li>
    <li>libsForQt5.kxmlrpcclient</li>
    <li>libsForQt5.kzones</li>
    <li>libsForQt5.layer-shell-qt</li>
    <li>libsForQt5.libgravatar</li>
    <li>libsForQt5.libkcddb</li>
    <li>libsForQt5.libkdegames</li>
    <li>libsForQt5.libkdepim</li>
    <li>libsForQt5.libkgapi</li>
    <li>libsForQt5.libkipi</li>
    <li>libsForQt5.libkleo</li>
    <li>libsForQt5.libkmahjongg</li>
    <li>libsForQt5.libkomparediff2</li>
    <li>libsForQt5.libksane</li>
    <li>libsForQt5.libkscreen</li>
    <li>libsForQt5.libksieve</li>
    <li>libsForQt5.libksysguard</li>
    <li>libsForQt5.libktorrent</li>
    <li>libsForQt5.lightly</li>
    <li>libsForQt5.mailcommon</li>
    <li>libsForQt5.mailimporter</li>
    <li>libsForQt5.marble</li>
    <li>libsForQt5.maui-core</li>
    <li>libsForQt5.mauikit-accounts</li>
    <li>libsForQt5.mauikit-calendar</li>
    <li>libsForQt5.mauikit-filebrowsing</li>
    <li>libsForQt5.mauikit-imagetools</li>
    <li>libsForQt5.mauikit-texteditor</li>
    <li>libsForQt5.mbox-importer</li>
    <li>libsForQt5.messagelib</li>
    <li>libsForQt5.milou</li>
    <li>libsForQt5.neochat</li>
    <li>libsForQt5.okular</li>
    <li>libsForQt5.oxygen</li>
    <li>libsForQt5.oxygen-sounds</li>
    <li>libsForQt5.palapeli</li>
    <li>libsForQt5.parachute</li>
    <li>libsForQt5.picmi</li>
    <li>libsForQt5.pim-data-exporter</li>
    <li>libsForQt5.pim-sieve-editor</li>
    <li>libsForQt5.pimcommon</li>
    <li>libsForQt5.plank-player</li>
    <li>libsForQt5.plasma-applet-caffeine-plus</li>
    <li>libsForQt5.plasma-applet-virtual-desktop-bar</li>
    <li>libsForQt5.plasma-bigscreen</li>
    <li>libsForQt5.plasma-browser-integration</li>
    <li>libsForQt5.plasma-desktop</li>
    <li>libsForQt5.plasma-dialer</li>
    <li>libsForQt5.plasma-disks</li>
    <li>libsForQt5.plasma-framework</li>
    <li>libsForQt5.plasma-integration</li>
    <li>libsForQt5.plasma-mobile</li>
    <li>libsForQt5.plasma-nano</li>
    <li>libsForQt5.plasma-nm</li>
    <li>libsForQt5.plasma-pa</li>
    <li>libsForQt5.plasma-remotecontrollers</li>
    <li>libsForQt5.plasma-sdk</li>
    <li>libsForQt5.plasma-settings</li>
    <li>libsForQt5.plasma-systemmonitor</li>
    <li>libsForQt5.plasma-thunderbolt</li>
    <li>libsForQt5.plasma-vault</li>
    <li>libsForQt5.plasma-wayland-protocols</li>
    <li>libsForQt5.plasma-welcome</li>
    <li>libsForQt5.plasma-workspace</li>
    <li>libsForQt5.plasma-workspace-wallpapers</li>
    <li>libsForQt5.plasmatube</li>
    <li>libsForQt5.polkit-kde-agent</li>
    <li>libsForQt5.powerdevil</li>
    <li>libsForQt5.print-manager</li>
    <li>libsForQt5.qqc2-breeze-style</li>
    <li>libsForQt5.qqc2-desktop-style</li>
    <li>libsForQt5.qtcurve</li>
    <li>libsForQt5.rocs</li>
    <li>libsForQt5.sddm-kcm</li>
    <li>libsForQt5.skanlite</li>
    <li>libsForQt5.skanpage</li>
    <li>libsForQt5.soundkonverter</li>
    <li>libsForQt5.spacebar</li>
    <li>libsForQt5.spectacle</li>
    <li>libsForQt5.station</li>
    <li>libsForQt5.systemsettings</li>
    <li>libsForQt5.tokodon</li>
    <li>libsForQt5.umbrello</li>
    <li>libsForQt5.xdg-desktop-portal-kde</li>
    <li>libsForQt5.yakuake</li>
    <li>libsForQt5.zanshin</li>
    <li>lightly-qt</li>
    <li>ludusavi</li>
    <li>lxqt.lxqt-panel</li>
    <li>lxqt.lxqt-powermanagement</li>
    <li>maia-icon-theme</li>
    <li>mangareader</li>
    <li>massif-visualizer</li>
    <li>material-kwin-decoration</li>
    <li>maui-shell</li>
    <li>nextcloud-client</li>
    <li>okteta</li>
    <li>partition-manager</li>
    <li>peruse</li>
    <li>plasma-applet-volumewin7mixer</li>
    <li>plasma-pass</li>
    <li>plasma-theme-switcher</li>
    <li>playbar2</li>
    <li>quassel</li>
    <li>quasselClient</li>
    <li>redshift-plasma-applet</li>
    <li>remmina</li>
    <li>rsibreak</li>
    <li>semantik</li>
    <li>sierra-breeze-enhanced</li>
    <li>skrooge</li>
    <li>syncthingtray</li>
    <li>systemdgenie</li>
    <li>tellico</li>
    <li>virt-manager-qt</li>
    <li>wacomtablet</li>
    <li>zombietrackergps</li>
  </ul>
</details>

File: nixos/modules/services/x11/desktop-managers/plasma5.nix [hunk 3]
```diff
@@ -387,7 +387,8 @@ in
         ++ lib.optional config.services.colord.enable pkgs.colord-kde
         ++ lib.optional config.services.hardware.bolt.enable pkgs.plasma5Packages.plasma-thunderbolt
         ++ lib.optionals config.services.samba.enable [ kdenetwork-filesharing pkgs.samba ]
-        ++ lib.optional config.services.xserver.wacom.enable pkgs.wacomtablet;
+        ++ lib.optional config.services.xserver.wacom.enable pkgs.wacomtablet
+        ++ lib.optional config.services.flatpak.enable pkgs.flatpak-kcm;
```
Comment (andrevmatos) [inline nixos/modules/services/x11/desktop-managers/plasma5.nix:391]: `flatpak-kcm` isn't in the root pkgs

```suggestion
        ++ lib.optional config.services.flatpak.enable pkgs.plasma5Packages.flatpak-kcm;
```

Comment (K900) [reply nixos/modules/services/x11/desktop-managers/plasma5.nix]: Nice catch, fixed.

**Comment by hqurve** [comment]
Thanks. I tested on my system with X11 and applied this patch using flake-utils-plus. After building, I rebooted into the upgraded generation

So far, I dont really see new issues. However, I noted the following

- the tiling overlay does not appear to work. After pressing Meta+T (the default keybinding), nothing happens and the following journal error occurs:
  ```
  ❯ journalctl --since "10 seconds ago" --no-pager
  Jan 24 19:12:30 JOSHUA-PC kwin_x11[2150]: Failed to load QUrl("file:///nix/store/iqfvx0xy1bwfyan7ymnz751b2aaqf04f-kwin-5.26.90/share/kwin/effects/tileseditor/qml/main.qml"): (file:///nix/store/iqfvx0xy1bwfyan7ymnz751b2aaqf04f-kwin-5.26.90/share/kwin/effects/tileseditor/qml/main.qml:15:1: module "org.kde.KWin.Effect.WindowView" is not installed
  Jan 24 19:12:30 JOSHUA-PC kwin_x11[2150]:     import org.kde.KWin.Effect.WindowView 1.0
  Jan 24 19:12:30 JOSHUA-PC kwin_x11[2150]:     ^)
  ```
  Note that if I tile windows as in the screenshot below, dragging the boundaries causes the relevant windows to autoresize (in a similar manner to Windows). So, I think it is just the overlay which is not working.

- The flatpak permission editor does show anything. However, discover appears to show the list of installed apps properly.
Screenshot from 2023-01-24 19-28-46


Regarding the plasma-welcome package, .... its only 1.8MiB. Personally, it is not that interesting to me since it just shows an autogenerated link to release notes. Still it is nice to see when plasma has updated if youre not actively checking.
Screenshot from 2023-01-24 19-39-55

**Comment by K900** [comment]
That's weird, the tiling overlay works just fine for me (both Wayland and X11). I'll check flatpak-kcm.

**Comment by K900** [comment]
image

flatpak-kcm also works...

**Comment by hqurve** [comment]
> That's weird, the tiling overlay works just fine for me (both Wayland and X11). I'll check flatpak-kcm.

By creating a "new user" (I `rm -r`'d a test account's home directory), I am also able to get the tiling overlay to work. ... I should have tested before/after too. I still have to figure out why it is not working on my main user. (I tried removing some of the cache directories, but it made no difference)

For the flatpak-kcm, I tried installing an application on the new user but nothing showed up. I then realized that it may be due to user(/system) remotes(/packages). On my main user, I setup a flatpak remote for the system and then installed a few packages. They all showed up in the settings. (I'm going to guess that this is an upstream bug)

<details>

Screenshot from 2023-01-25 07-29-35

</details>

**Comment by oxalica** [comment]
I've been using this (rebased to nixos-unstable) for some days and it's working fine on wayland. The new fractional scaling support also works well with `alacritty` master.

**Review by samueldr — CHANGES_REQUESTED** [review]
Doesn't apply currently, is still needed on top of yesterday's `nixos-unstable`:

 - https://github.com/NixOS/nixpkgs/blob/f2a2c2dbaabab418702fa2bc39f503a1f6efe59e/pkgs/desktops/plasma-5/kwin/default.nix#L145-L146

(Will likely come back to it soon~ish, once I'm done with the higher priority task.)

* * *

The patch applies mostly to the new file replacing it (the hunk that doesn't isn't there anymore).

Though it is not enough, in the meantime `src/libkwineffects/kwineffects.h` also grew modern C++ usage of `<span>`, and is not one that can *just* be reverted away.

**Comment by K900** [comment]
For multiscreen users: https://notmart.org/blog/2023/02/how-to-report-multiscreen-bugs/

**Comment by peterhoeg** [comment]
I'm aware that this comment is just noise, but I *cannot* wait for this to land as multi-screen on my laptop through USB type C docks has been strange for a while.

**Comment by K900** [comment]
Should be good to go, running one last nixpkgs-review to confirm first.

**Comment by K900** [comment]
Result of `nixpkgs-review pr 211767` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>93 packages built:</summary>
  <ul>
    <li>breath-theme</li>
    <li>digikam</li>
    <li>goverlay</li>
    <li>haruna</li>
    <li>kde-rounded-corners</li>
    <li>libsForQt5.applet-window-buttons</li>
    <li>libsForQt5.aura-browser</li>
    <li>libsForQt5.bismuth</li>
    <li>libsForQt5.bluedevil</li>
    <li>libsForQt5.breeze-grub</li>
    <li>libsForQt5.breeze-gtk</li>
    <li>libsForQt5.breeze-plymouth</li>
    <li>libsForQt5.breeze-qt5</li>
    <li>libsForQt5.buho</li>
    <li>libsForQt5.clip</li>
    <li>libsForQt5.communicator</li>
    <li>libsForQt5.discover</li>
    <li>libsForQt5.flatpak-kcm</li>
    <li>libsForQt5.kactivitymanagerd</li>
    <li>libsForQt5.kde-cli-tools</li>
    <li>libsForQt5.kde-gtk-config</li>
    <li>libsForQt5.kde2-decoration</li>
    <li>libsForQt5.kdecoration</li>
    <li>libsForQt5.kdenlive</li>
    <li>libsForQt5.kdeplasma-addons</li>
    <li>libsForQt5.kdev-php</li>
    <li>libsForQt5.kdev-python</li>
    <li>libsForQt5.kdevelop</li>
    <li>libsForQt5.kdevelop-unwrapped</li>
    <li>libsForQt5.kgamma5</li>
    <li>libsForQt5.khotkeys</li>
    <li>libsForQt5.kinfocenter</li>
    <li>libsForQt5.kmenuedit</li>
    <li>libsForQt5.kpipewire</li>
    <li>libsForQt5.krohnkite</li>
    <li>libsForQt5.kscreen</li>
    <li>libsForQt5.kscreenlocker</li>
    <li>libsForQt5.ksshaskpass</li>
    <li>libsForQt5.ksystemstats</li>
    <li>libsForQt5.kuserfeedback</li>
    <li>libsForQt5.kwallet-pam</li>
    <li>libsForQt5.kwayland-integration</li>
    <li>libsForQt5.kwin</li>
    <li>libsForQt5.kwin-dynamic-workspaces</li>
    <li>libsForQt5.kwin-tiling</li>
    <li>libsForQt5.kwrited</li>
    <li>libsForQt5.kzones</li>
    <li>libsForQt5.layer-shell-qt</li>
    <li>libsForQt5.libkscreen</li>
    <li>libsForQt5.libksysguard</li>
    <li>libsForQt5.lightly</li>
    <li>libsForQt5.milou</li>
    <li>libsForQt5.nota</li>
    <li>libsForQt5.oxygen</li>
    <li>libsForQt5.oxygen-sounds</li>
    <li>libsForQt5.parachute</li>
    <li>libsForQt5.pix</li>
    <li>libsForQt5.plank-player</li>
    <li>libsForQt5.plasma-bigscreen</li>
    <li>libsForQt5.plasma-browser-integration</li>
    <li>libsForQt5.plasma-desktop</li>
    <li>libsForQt5.plasma-disks</li>
    <li>libsForQt5.plasma-integration</li>
    <li>libsForQt5.plasma-mobile</li>
    <li>libsForQt5.plasma-nano</li>
    <li>libsForQt5.plasma-nm</li>
    <li>libsForQt5.plasma-pa</li>
    <li>libsForQt5.plasma-remotecontrollers</li>
    <li>libsForQt5.plasma-sdk</li>
    <li>libsForQt5.plasma-systemmonitor</li>
    <li>libsForQt5.plasma-thunderbolt</li>
    <li>libsForQt5.plasma-vault</li>
    <li>libsForQt5.plasma-welcome</li>
    <li>libsForQt5.plasma-workspace</li>
    <li>libsForQt5.plasma-workspace-wallpapers</li>
    <li>libsForQt5.polkit-kde-agent</li>
    <li>libsForQt5.powerdevil</li>
    <li>libsForQt5.qqc2-breeze-style</li>
    <li>libsForQt5.sddm-kcm</li>
    <li>libsForQt5.shelf</li>
    <li>libsForQt5.spectacle</li>
    <li>libsForQt5.systemsettings</li>
    <li>libsForQt5.vvave</li>
    <li>libsForQt5.xdg-desktop-portal-kde</li>
    <li>lightly-qt</li>
    <li>lxqt.lxqt-config</li>
    <li>maliit-framework</li>
    <li>maliit-keyboard</li>
    <li>material-kwin-decoration</li>
    <li>maui-shell</li>
    <li>plasma-applet-volumewin7mixer</li>
    <li>sierra-breeze-enhanced</li>
    <li>wacomtablet</li>
  </ul>
</details>

**Comment by K900** [comment]
image

Works great here.

File: pkgs/desktops/plasma-5/kinfocenter/default.nix [hunk 4]
```diff
@@ -99,6 +99,6 @@ mkDerivation {
   # the same directory, while it is actually located in a completely different
   # store path
   preFixup = ''
-    ln -sf ${lib.getExe systemsettings} $out/bin/kinfocenter
+    ln -sf ${systemsettings}/bin/systemsettings $out/bin/kinfocenter
```
Comment (oxalica) [inline pkgs/desktops/plasma-5/kinfocenter/default.nix:102]: Just curious: does `getExe` break anything here?

Comment (K900) [reply pkgs/desktops/plasma-5/kinfocenter/default.nix:102]: Yes, it picks `systemsettings5` which is a symlink that ends up double wrapped.

Comment (peterhoeg) [reply pkgs/desktops/plasma-5/kinfocenter/default.nix:102]: Then we probably should do `meta.mainProgram = "systemsettings";`

Comment (K900) [reply pkgs/desktops/plasma-5/kinfocenter/default.nix:102]: What we should really do is to fix wrapQtAppsHook so it doesn't touch symlinks.

Comment (andrevmatos) [reply pkgs/desktops/plasma-5/kinfocenter/default.nix:102]: Cool, `About this System` menu entry works fine now (before, it just opened System Settings home screen)

**Review by andrevmatos — APPROVED** [review]
working fine here

**Review by samueldr — APPROVED** [review]
Tested Plasma Mobile with this PR on top of NixOS unstable on AArch64, on `lenovo-wormdingler`.

**Pull request merged**